fr.esrf.tango.generator.xtend/src/fr/esrf/tango/pogo/generator/python/PythonDeviceHL.xtend

```diff
@@ -208,15 +208,38 @@ __all__ = ["«cls.name»", "main"]
     //    Constructors
     //====================================================
     def pythonConstructors(PogoDeviceClass cls)  '''
-«IF true»«cls.constructorHL»
+«IF true»    def init_device(self):
+        """ 
+        Initializes the attributes and properties of the «cls.name»
+        """
+«cls.constructorHL»
         «cls.setEventCriteria»
-        «IF cls.description.filestogenerate.toLowerCase.contains("protected regions")»«cls.protectedAreaHL("init_device")»«ENDIF»
+        «IF cls.description.filestogenerate.toLowerCase.contains("protected regions")»
+        «cls.openProtectedAreaHL("init_device")»
+        «cls.closeProtectedAreaHL("init_device")»
+        «ENDIF»
 
     def always_executed_hook(self):
-        «IF cls.description.filestogenerate.toLowerCase.contains("protected regions")»«cls.protectedAreaHL("always_executed_hook", "pass", false)»«ELSE»pass«ENDIF»
+        «IF cls.description.filestogenerate.toLowerCase.contains("protected regions")»
+        «cls.openProtectedAreaHL("always_executed_hook")»
+        """
+        Internal construct of TANGO.
+        """
+        «cls.closeProtectedAreaHL("always_executed_hook")»
```

Comment:
Shouldn't this docstring be before the protected area?  Similar to `init_device`, we don't expect the user to modify it, so does not need to be preserved between code generation runs.

Reply:
Agree with the suggestion. Removed docstrings from protected region for constructor methods. Link to the commit on fork branch:
https://github.com/ska-telescope/pogo/commit/d07fba94c24464933569ea6272071cb4d3fa4ff6